lib/awful/key.lua

```diff
@@ -1,6 +1,51 @@
 ---------------------------------------------------------------------------
 --- Create easily new key objects ignoring certain modifiers.
 --
+-- A key object can be used by @{awful.keyboard} and @{client} to define
+-- keybindings.
+--
+-- Use awful.key to define a keybinding
+-- ---
+--
+-- This example shows how to define a basic key object:
+--
+--    awful.key({ "Mod4", "Shift" }, 'a',
+--      function () print("The `Mod4` + `Shift` + `a` combo is pressed") end,
+--      function () print("The `Mod4` + `Shift` + `a` combo is released") end)
+--
+-- This example shows how to define the same basic key object with the
+-- declarative pattern:
+--
+--    awful.key {
+--      modifiers = { "Mod4", "Shift" },
+--      key = 'a',
+--      on_press = function ()
+--         print("The `Mod4` + `Shift` + `a` combo is pressed")
+--      end,
+--      on_release = function ()
+--         print("The `Mod4` + `Shift` + `a` combo is released")
+--      end
+--    }
```

Comment:
would be better to add `description` field to the example

Reply:
I wasn't totally sure how to document this part. `awful.key` can "save" additional parameters as data-fields. Keybindings uses them for the `description` and `group` data-fields. But other than that, it's not really a part of the key API...

Hm... Actually, you're right, we should have a mention for this anyway. I'll add that!

Reply:
If you get some time, can you move that code to `tests/examples/text/`? This way (hopefully), it gets executed and make the build fail if there are syntax errors.

Reply:
Another notable benefice is that the indentation levels are easier to make right without the leading `--` + extra spaces from ldoc code block.